File: math/0754_reach_a_number/0754_reach_a_number.py
class Solution:
    def reachNumber(self, target: int) -> int:
        target = abs(target)
        p, i = 0, 0
        while p < target or (p + target) % 2 != 0:
            i += 1
            p = p + i
        return i


target = 1
res = Solution().reachNumber(target)
print(res)

File: math/0754_reach_a_number/0754_reach_a_number.go
package main

import "fmt"

func reachNumber(target int) int {
	if target < 0 {
		target = -target
	}
	for n := 1; ; n++ {
		sub := n*(n+1)/2 - target
		if sub >= 0 && sub&1 == 0 {
			return n
		}
	}
}

func main() {
	target := 1
	res := reachNumber(target)
	fmt.Println(res)
}
